index.js

```diff
@@ -115,7 +129,7 @@ exports.search = utils.toPromise(function (opts, callback) {
   }
 
   var fieldBoosts = Object.keys(fields).map(function (field) {
-    var deepField = field.indexOf('.') !== -1 && field.split('.');
+    var deepField = field.indexOf('.') !== -1 && field.replace(/\[\]/g, '.').split('.');
```

Comment:
This is a neat idea, but is there any reason we can't have `foo.bar.baz` instead of `foo.bar[].baz`? Making the user specify what is and isn't an array seems unnecessary, since we can just test `typeof text` and `Array.isArray(text)` to see if it's a string, object, or array, and act accordingly.